opaque_keys/edx/locations.py

```diff
@@ -209,9 +209,9 @@ class DeprecatedLocation(BlockUsageLocator):
     The short-lived location:org+course+run+block_type+block_id syntax
     """
     CANONICAL_NAMESPACE = 'location'
-    URL_RE_SOURCE = r"""
-        (?P<org>{ALLOWED_ID_CHARS}+)\+(?P<course>{ALLOWED_ID_CHARS}+)\+(?P<run>{ALLOWED_ID_CHARS}+)\+
-        (?P<block_type>{ALLOWED_ID_CHARS}+)\+
+    URL_RE_SOURCE = u"""
```

Comment:
There are backslashes in the ALLOWED_ID_CHARS var, do those need to be updated as well?

Reply:
No, the `r` at the start of a string literal only affects how characters shown explicitly in the source code for that string literal are treated.  (The combined `ur` prefix no longer exists in Python 3, hence why we need the escaping here).

Reply:
ok, so the contents of the var will be escaped correctly before the strings are combined?

Reply:
Yes, its content is independent of the prefixes used for the string literal referencing it.